.gitignore

```diff
@@ -1,37 +1,2 @@
-# See http://help.github.com/ignore-files/ for more about ignoring files.
-
-# compiled output
-/dist
-
-# dependencies
-/node_modules
-
-# IDEs and editors
-/.idea
-.project
-.classpath
-.c9/
-*.launch
-.settings/
-*.sublime-workspace
-
-# IDE - VSCode
-.vscode/*
-!.vscode/settings.json
-!.vscode/tasks.json
-!.vscode/launch.json
-!.vscode/extensions.json
-
-# misc
-/.sass-cache
-/connect.lock
-/coverage
-/libpeerconnection.log
-npm-debug.log
-yarn-error.log
-testem.log
-/typings
-
-# System Files
-.DS_Store
-Thumbs.db
+coverage/
+node_modules/
```

Comment:
I'm not sure how publishing will look like, however, attaching `dist` folder is important if someone is adding library via git path. I believe it won't change for `npm` publishing, so `dist` package should be also pushed into a repository and updated when needed. The same has been done for CKEditor5 Vue integration - https://github.com/ckeditor/ckeditor5-vue and is done for most popular Vue plugins, e.g. see `vue-router` - https://github.com/vuejs/vue-router

Reply:
However our React integration, `ckeditor4-react`, does not include `dist` directory – even for tags. Probably we can rethink the whole distribution system. I'm for pushing `dist` directory only for tags (similar to e.g. `rollup-plugin-babel-minify`). `vue-router` has always `dist` directory, but it seems to be updated only for tags – so it won't be helpful for adding library via git.

Reply:
>  I'm for pushing dist directory only for tags (similar to e.g. rollup-plugin-babel-minify)

What do you mean about tags? :thinking: 

I'm fine for removing `dist` folder for dev version as using git path is not necessary a valid way to include a plugin (especially for production version), however, how would be `dist` folder published by npm? By adding it to `files` option and building + publishing on every release?

Reply:
> What do you mean about tags?

https://github.com/Comandeer/rollup-plugin-babel-minify/blob/90ca4099d88387fce7eb39f8364a0ce1036250f5/package.json#L18-L19

> how would be dist folder published by npm? By adding it to files option and building + publishing on every release?

Yes.

Reply:
Ok, it looks like the best option for me :+1:  Also this 'way' of versioning is described in npm docs (https://docs.npmjs.com/cli/version) if someone needs more depth documentation.

Reply:
I'm not sure if we need `dist/` in the main repo? What do you mean by:

> if someone is adding library via git path

is it a case when one just clones the repo and uses it? I think in this case `dist/` should be simply build by the person doing that as he is using dev (not production/dist) version of the integration. And if you are talking about using repo as node module I would be more for adding build step as a postinstall step.
And as for tags - since tags never change one also may build `dist/` after cloning the tag (I assume it's one command, right?).

Reply:
> is it a case when one just clones the repo and uses it?

no, it's a case when you are using github as a npm package path, e.g.

```
dependencies: {
  "ckeditor4-vue": "https://github.com/ckeditor/ckeditor4-vue.git#1.0.0"
}
```
but as I understand, versioning proposed by @Comandeer will take care of keeping `dist` folders in git tag.

`dist` folder is required for npm to work, otherwise requiring plugin from node won't load correctly (e.g. `import CKEditor4 from 'ckeditor4-vue` will throw error).

Reply:
Yes, I think that's what @Comandeer had in mind. Can't we make building (so creating `dist/`) a post-install step then? WDYT @Comandeer @jacekbogdanski 
Not sure what would be the benefit apart from not having `dist/` folder (which was the main discussion point here) :thinking:

Reply:
A simple post-install script should also probably work here, so `dist` is produced only for npm installation. :+1: from me (if it will work). Both options are fine - with post-install we are losing dist tags versioning when using gitpath, but it's probably not really important here.

I don't have a really hard opinion on which one method is better, but considering that @f1ames one seems less complicated, I'm probably slightly more for it. @Comandeer do you have some arguments which could help choose using versioning script with git tags over simplified @f1ames idea?

Reply:
Hmmm, so user installing package from npm will get `dist/`, but the package will rebuild it due to `postinstall` script?

Reply:
Hmmm, good point, I was thinking only about rebuilding in postinstall when github repo is used as NPM package, not a dist package - however this again will add some complexity to the process as we will need different flow for dist version (no postinstall step) :thinking:

Reply:
So, @Comandeer way than? It seems legit.